static/js/input.js

```diff
@@ -446,42 +445,32 @@ function guessColumnType(arr, threshold = 600) {
 
 }
 
-
 /**
  * Calculates the entropy of an array
  * @function calcEntropy
  * @param {Array} arr - array of values
  * @returns {Number} entropy - entropy of array
  */
-function calcEntropy(arr, match) {
-  const unique = [...new Set(arr)];
-  const base = unique.length;
-  let probs = [];
-
-  // calculates probability of occurence of search value
-  function calcProb(match) {
-    const n = arr.length;
-
-    let count = 0;
-    for (let i = 0; i < n; i++) {
-      if (arr[i] === match) {
-        count++;
-      }
+function calcEntropy(arr) {
+  let unique = [],
+      probs = [];
+
+  for (let i = 0; i < arr.length; i++) {
+    if (unique.length === 0 || unique.indexOf(arr[i]) < 0) {
+      unique.push(arr[i]);
+      probs.push(1);
+    }
+    else {
+      probs[unique.indexOf(arr[i])] = probs[unique.indexOf(arr[i])] + 1;
     }
-
-    return count / n;
   }
 
-  for (let val of unique) probs.push(calcProb(val));
-
-  const reducer = (acculumator, current) => acculumator + (current + (Math.log(current) / Math.log(base)));
-  const entropy = -1 * probs.reduce(reducer);
-
-  return entropy;
+  let entropy = 0;
+  for (let i = 0; i < probs.length; i++) entropy += (probs[i] / arr.length) * Math.log2(probs[i] / arr.length);
```

Comment:
If a line is longer than 80 characters, can break that down.

Can do `entropy -= ...` since you want a negative value.

`probs[i] / arr.length` is calculated twice. So it will be better to calculate it once and store in an intermediate variable in the loop, then...